scripts/rollup/build.js

```diff
@@ -561,9 +573,32 @@ function getPlugins(
           // Because it sits on disk, there's no need to have it in the `sourcesContent` array.
           // That also makes the file easier to read, and available for use by scripts.
           // This should be the only file in the array.
-          const [preMinifiedBundleSource] =
+          const [preMinifiedBundleSourceWithoutLicense] =
             sourcemapAfterClosure.sourcesContent;
 
+          // We were passing sources without license headers to the Closure compiler.
+          // The reason for it is its optimization for multiline comment:
+          // it will change the format by removing * prefixes This doesn't work well with Haste.
+          const preMinifiedBundleSource = Wrappers.wrapWithLicenseHeader(
```

Comment:
:question: Hmm. I _am_ curious if the sourcemap transforms are still going to be accurate vs the output file contents if we don't include the license contents before the sourcemap gets generated.

This is an area where my own knowledge is very hazy.  My coworker @hbenl _might_ have more knowledge here.

My concern is that sending one version of the source file through Closure, and writing a modified file to disk, might cause line numbers to get mismatched when it's used in a debugger (such that it thinks you're paused on line 37 of a file, but that turns out to be a `}` character or something instead of the actual meaningful line of code).

I'd recommend doing some checks with the artifacts from this PR to see if setting breakpoints and stepping still seems to result in correct lines and variables being handled.

Reply:
I've ended up with manually updating the header to follow certain format. I couldn't find any annotations for Closure compiler that would allow us to preserve original header-comment format, unfortunately.

I've validated this in my template React-app, here is the demo, where I've replaced `react-dom` package to the one with sourcemaps:

https://github.com/facebook/react/assets/28902667/e198218d-9918-4643-8c9c-e8b9a5012434

Reply:
Alright, I think I found the best option for us:

1. Artifacts with sourcemaps will get header attached before, so it will have an effect on the sourcemaps and they will be generated correctly.
2. Artifacts without sourcemaps (facebook-bundles are included in this category) will get header attached after Closure compiler, so they will preserve correct format of the header.

I've validated it the same way as above, sourcemaps work.